lib/CL/devices/spirv_parser.cc

```diff
@@ -35,32 +36,47 @@
 #include "spirv_parser.hh"
 
 #include "pocl_debug.h"
+#include "pocl_util.h"
 
 #define logWarn(...) POCL_MSG_WARN(__VA_ARGS__);
 #define logError(...) POCL_MSG_ERR(__VA_ARGS__);
 #define logTrace(...) POCL_MSG_PRINT_INFO(__VA_ARGS__);
 
-typedef std::map<int32_t, std::shared_ptr<OCLFuncInfo>> OCLFuncInfoMap;
+/// Reinterpret the pointed region, starting from BaseAddr +
```

Comment:
AFAICT `reinterpret_cast` + a load could break if the alignment requirement of the output is larger than that of input, whereas `memcpy` generates unaligned copy instructions which always work. Since this is a template, `memcpy` has to be used.